js/assessment.js

```diff
@@ -225,10 +229,21 @@ define([
             if (this._assessments._byPageId[pageId] === undefined) {
                 this._assessments._byPageId[pageId] = [];
             }
+
             this._assessments._byPageId[pageId].push(assessmentModel);
 
-            if (assessmentId) {
-                this._assessments._byAssessmentId[assessmentId] = assessmentModel;
+            if (assessmentId !== undefined) {
+                if (assessmentId === '') {
+                    Adapt.log.warn("An assessment has been registered with an empty value for 'id'");
+                }
+
+                if (!this._assessments._byAssessmentId[assessmentId]) {
+                    this._assessments._byAssessmentId[assessmentId] = assessmentModel;
+                } else {
+                    Adapt.log.warn("An assessment with an id of '" + assessmentId + "' already exists!");
+                }
+            } else {
+                Adapt.log.warn("An assessment has been registered with an undefined value for 'id'");
             }
```

Comment:
@oliverfoster, think the idea was to allow an empty string ID, `""`,  but not `undefined`.

To banish the nesting, you could well split it into a separate function e.g.
```js
this._addToAssessmentIdMap(assessmentId, assessmentModel);
```
```js
_addToAssessmentIdMap: function(id, model) {
    if (id === undefined) {
        Adapt.log.warn("An assessment has been registered with an undefined value for 'id'");
        return;
    }

    if (id === '') {
        Adapt.log.warn("An assessment has been registered with an empty value for 'id'");
    }

    if (!this._assessments._byAssessmentId[id]) {
        this._assessments._byAssessmentId[id] = model;
    } else {
        Adapt.log.warn("An assessment with an id of '" + id + "' already exists!");
    }
}
```